src/renderer/hooks/usePaneNodeSearchMenu.tsx

```diff
@@ -0,0 +1,299 @@
+import { CloseIcon, SearchIcon } from '@chakra-ui/icons';
+import {
+    Box,
+    HStack,
+    Input,
+    InputGroup,
+    InputLeftElement,
+    InputRightElement,
+    MenuList,
+    Text,
+    useColorModeValue,
+} from '@chakra-ui/react';
+import log from 'electron-log';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { Node, OnConnectStartParams, useReactFlow } from 'react-flow-renderer';
+import { useContext } from 'use-context-selector';
+import { NodeData, NodeSchema } from '../../common/common-types';
+import { createUniqueId, parseHandle } from '../../common/util';
+import { IconFactory } from '../components/CustomIcons';
+import { ContextMenuContext } from '../contexts/ContextMenuContext';
+import { GlobalContext, GlobalVolatileContext, NodeProto } from '../contexts/GlobalNodeState';
+import getNodeAccentColors from '../helpers/getNodeAccentColors';
+import { getMatchingNodes, getNodesByCategory } from '../helpers/nodeSearchFuncs';
+import { useContextMenu } from './useContextMenu';
+
+interface UsePaneNodeSearchMenuValue {
+    readonly onConnectStart: (event: React.MouseEvent, handle: OnConnectStartParams) => void;
+    readonly onConnectStop: (event: MouseEvent) => void;
+    readonly onPaneContextMenu: (event: React.MouseEvent) => void;
+}
+
+export const usePaneNodeSearchMenu = (
+    wrapperRef: React.RefObject<HTMLDivElement>
+): UsePaneNodeSearchMenuValue => {
+    const { createNode, createConnection } = useContext(GlobalVolatileContext);
+    const { closeContextMenu } = useContext(ContextMenuContext);
+    const { schemata } = useContext(GlobalContext);
+
+    const [connectingFrom, setConnectingFrom] = useState<OnConnectStartParams | null>();
+    const [connectingFromType, setConnectingFromType] = useState<string | null>();
+    const [isStoppedOnPane, setIsStoppedOnPane] = useState<boolean>(false);
```

Comment:
The type can be inferred :)

```suggestion
    const [isStoppedOnPane, setIsStoppedOnPane] = useState(false);
```